### PR TITLE
Pfe navigation light dom

### PR DESCRIPTION
**Pull request by LyndseyR**
## Title of this pull request

* Link to discovery documentation (should contain all the planning and requirements for this work):
   - See the `DISCOVERY.md` template at the root of the project
* Link(s) to demo pages where this element can be viewed:
    - http://localhost:8000/elements/pfe-navigation/demo/
---

### What has changed and why
Summarize files edited as part of this MR along with a brief description of what was changed/why.

* pfe-navigation/demo/index.html - making this match the mockups to be shared out
* pfe-navigation/src/pfe-navigation--lightdom-2.scss - making temporary file that will merge over into --lightdom.scss when pfe-navigation branch is merged in to master. 

### Testing instructions
Be sure to include detailed instructions on how your update can be tested by another developer.

1. 

#### Browser requirements
Your component should work in all of the following environments:

- [ ] Latest 2 versions of Edge
- [ ] Internet Explorer 11 (should be useable, not pixel perfect)
- [ ] Latest 2 versions of Firefox (one on Mac OS, one of Windows OS)
- [ ] Firefox 60.7.2 (or latest version for Red Hat Enterprise Linux distribution)
- [ ] Latest 2 versions of Chrome (one on Mac OS, one of Windows OS)
- [ ] Latest 2 versions of Safari
- [ ] Galaxy S9 Firefox
- [ ] iPhone X Safari
- [ ] iPad Pro Safari
- [ ] Pixel 3 Chrome

Your repository infrastructure updates should work for at least:
- [ ] Node v8.x
- [ ] NPM v7.x

### Ready-for-merge Checklist

- [ ] Expected files: all files in this pull request are related to one feature request or issue (no stragglers)?
- [ ] Did automated tests pass?
- [ ] Did browser testing pass?
- [ ] Did you update or add any necessary documentation (README.md, WHY.md, etc.)?
- [ ] Was this feature demo'd and the design review approved?
- [ ] Did it get a LGTM *after the last commit*?
- [ ] Did you update the CHANGELOG.md file with a summary of this update?

**Be sure to share your updates with the [patternfly-elements-contribute@redhat.com](mailto:patternfly-elements-contribute@redhat.com) mailing list!**

### CODE REVIEW

File: elements/pfe-navigation/demo/index.html [hunk 1]
```diff
@@ -70,95 +71,410 @@ <h2 slot="mobile-menu--label">Menu</h2>
...
-                          <a href="#">Platforms</a>
-                        </h3>
-                        <ul class="pfe-link-list--list">
-                          <li class="pfe-link-list--list-item"><a href="#">Red Hat Enterprise Linux</a></li>
-                          <li class="pfe-link-list--list-item"><a href="#">Red Hat JBoss Enterprise Application Platform</a></li>
-                          <li class="pfe-link-list--list-item"><a href="#">Red Hat OpenShift</a></li>
-                          <li class="pfe-link-list--list-item"><a href="#">Red Hat OpenStack Platform</a></li>
-                          <li class="pfe-link-list--list-item"><a href="#">Red Hat Virtualization</a></li>
-                        </ul>
+                <p slot="trigger"><a href="#">Products</a></p>
```
[the remark this answers is not in the document]
Comment (LyndseyR) [reply elements/pfe-navigation/demo/index.html]: I tried reopening a discussion in the handoff doc, https://docs.google.com/document/d/11EP_2s3TLuA5KuZfhj3QKXSa7vJGKQYJndJN9a4TAYU/edit#, Kendall asked Jarius and the recommendation was to make it a p tag. I can see the comment but the content that it highlighted is no longer there, so I might have misinterpreted but I'm fairly certain they intentionally wanted these as p's

File: elements/pfe-navigation/demo/index.html [hunk 2]
```diff
@@ -70,95 +71,410 @@ <h2 slot="mobile-menu--label">Menu</h2>
...
+                <div slot="tray" hidden>
+                  <div class="pfe-navigation-item__tray--container">
+                     <div class="pfe-nav-grid">
+                    <div class="column">
```
[the remark this answers is not in the document]
Comment (LyndseyR) [reply elements/pfe-navigation/demo/index.html]: I changed it to `pfe-navigation-column` does that make sense?

File: elements/pfe-navigation/demo/index.html [hunk 3]
```diff
@@ -43,9 +44,9 @@ <h1>&lt;pfe-navigation&gt;</h1>
         </ul>
       </div>
       <!-- Site logo -->
-      <h2 slot="logo">
+      <h2 slot="logo" style="padding: 0 0 8px 0;line-height: 1;">
         <a href="https://redhat.com">
-          <img class="redhat-logo" src="https://via.placeholder.com/150x50.png" class="screen" title="Red Hat"/>
+          <img src="/profiles/rh/themes/redhatdotcom/img/logo.svg" class="screen" title="Red Hat" style="width: 135px; height: 100%;">
```
[the remark this answers is not in the document]
Comment (LyndseyR) [reply elements/pfe-navigation/demo/index.html]: Roger!

File: elements/pfe-navigation/demo/index.html (hunk 2 above)
[the remark this answers is not in the document]
Comment (castastrophe) [reply elements/pfe-navigation/demo/index.html]: Would you mind using the BEM syntax: `pfe-navigation--column`?

File: elements/pfe-navigation/demo/index.html [hunk 4]
```diff
@@ -70,95 +71,410 @@ <h2 slot="mobile-menu--label">Menu</h2>
...
-                          <li class="pfe-link-list--list-item"><a href="#">Red Hat OpenStack Platform</a></li>
-                          <li class="pfe-link-list--list-item"><a href="#">Red Hat Virtualization</a></li>
-                        </ul>
+                <p slot="trigger"><a href="#">Products</a></p>
+                <div slot="tray" hidden>
+                  <div class="pfe-navigation-item__tray--container">
+                     <div class="pfe-nav-grid">
+                    <div class="column">
+                        <div class="pfe-link-list">
+                          <p class="pfe-link-list--header">Platforms</p>
```
[the remark this answers is not in the document]
Comment (markcaron) [reply elements/pfe-navigation/demo/index.html]: Can we also have it link up the header to the list of links using `aria-labelledby` when it upgrades? These IDs can be autogenerated if they don't exist in the lightDOM. I did something similar in the `pfe-footer` branch/element.

For example:

```
<h3 id="pfe-nav__h-cfgeh2343nsah373ja">Platforms</h3>
<ul aria-labelledby="pfe-nav__h-cfgeh2343nsah373ja">
...
</ul>
```

[the remark this answers is not in the document]
Comment (castastrophe) [reply elements/pfe-navigation/demo/index.html]: These are lightDOM-only styles that Lyndsey is working on so the web component won't be able to handle the addition of the aria-labelledby tag but we can manually add it to the markup and share that with the dev team (Drupal will be outputting this markup).

[the remark this answers is not in the document]
Comment (markcaron) [reply elements/pfe-navigation/demo/index.html]: Ok, right. Missed the "lightDOM" part of the criteria for this PR. Sounds good! When this becomes it's fully-fledged web component, we'll handle it in a similar way (JS upgrade).

[the remark this answers is not in the document]
Comment (castastrophe) [reply elements/pfe-navigation/demo/index.html]: @LyndseyR Let's do this via the analytics branch that's coming next sprint - don't worry about incorporating these aria labels yet.

**Comment by castastrophe** [comment]
👍 Thanks for this Lyndsey!  It looks great.  We'll resolve the testing errors in pfe-navigation branch.

**Pull request merged**